talk/objectorientation/advancedoo.tex

```diff
@@ -305,6 +306,54 @@
   \end{multicols}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitlecpp[98]{Polymorphism and destruction}
+  \begin{block}{Owning base pointers}
+    \begin{itemize}
+    \item We sometimes need to maintain owning pointers to base classes:
+    \begin{cppcode*}{gobble=2}
+      struct Drawable {
+         virtual void draw() = 0;
+      };
+
+      Drawable* p = getImpl(); // returns derived obj.
+      p->draw();
+      delete p;
+      // better interface (discussed later):
+      // std::unique_ptr<Drawable> p = getImpl();
+      // p->draw();
```

Comment:
```suggestion
```

Don't do this here. It's too much to read, and we're not there yet.
If you want to keep it, make it a thing that's being uncovered on the next slide -- à la:
BTW, this is how you do it today [...]

But even that might not be such a good style. 🤔

Reply:
Referring to the exampleblock comment above, you could just show the new exampleblock with the new code if you want to keep the comment about the unique_ptr.

Reply:
I put it onto a separate slide using `overprint`. What do you think?

Reply:
Better! 👍

Reply:
Great!